pkg/cmd/roachtest/tests/decommissionbench.go

```diff
@@ -0,0 +1,317 @@
+			warehouses:       1000,
+			load:             false,
+			admissionControl: false,
+		},
+	} {
+		registerDecommissionBenchSpec(r, tc)
+	}
+}
+
+// registerDecommissionBenchSpec adds a test using the specified configuration to the registry.
+func registerDecommissionBenchSpec(r registry.Registry, tc decommissionBenchSpec) {
+	timeout := defaultTimeout
+	extraNameParts := []string{""}
+
+	if tc.snapshotRate == 0 {
+		tc.snapshotRate = defaultSnapshotRateMb
+	} else {
+		extraNameParts = append(extraNameParts, fmt.Sprintf("snapshotRate=%dmb", tc.snapshotRate))
+	}
+
+	if tc.whileDown {
+		extraNameParts = append(extraNameParts, "while-down")
+	}
+
+	if !tc.load {
+		extraNameParts = append(extraNameParts, "no-load")
+	}
+
+	if !tc.admissionControl {
+		extraNameParts = append(extraNameParts, "no-admission")
+	}
+
+	extraName := strings.Join(extraNameParts, "/")
+
+	r.Add(registry.TestSpec{
+		Name: fmt.Sprintf("decommissionBench/nodes=%d/cpu=%d/warehouses=%d%s",
+			tc.nodes, tc.cpus, tc.warehouses, extraName),
+		Owner:             registry.OwnerKV,
+		Cluster:           r.MakeClusterSpec(tc.nodes+1, spec.CPU(tc.cpus)),
+		Timeout:           timeout,
+		NonReleaseBlocker: true,
+		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+			runDecommissionBench(ctx, t, c, tc, timeout)
+		},
+	})
+}
+
+// decommBenchTicker is a simple struct for encapsulating function
+// closures to be called before and after an operation to be benchmarked.
+type decommBenchTicker struct {
+	pre  func()
+	post func()
+}
+
+// runDecommissionBench initializes a cluster with TPCC and attempts to
+// benchmark the decommissioning of a single node picked at random. The cluster
+// may or may not be running under load.
+func runDecommissionBench(
+	ctx context.Context,
+	t test.Test,
+	c cluster.Cluster,
+	tc decommissionBenchSpec,
+	testTimeout time.Duration,
+) {
+	// node1 is kept pinned (i.e. not decommissioned/restarted), and is the node
+	// through which we run decommissions. The last node is used for the workload.
+	pinnedNode := 1
+	workloadNode := tc.nodes + 1
+	crdbNodes := c.Range(pinnedNode, tc.nodes)
+	c.Put(ctx, t.Cockroach(), "./cockroach", c.All())
+	c.Put(ctx, t.DeprecatedWorkload(), "./workload", c.Node(workloadNode))
+	for i := 1; i <= tc.nodes; i++ {
+		startOpts := option.DefaultStartOpts()
+		startOpts.RoachprodOpts.ExtraArgs = append(startOpts.RoachprodOpts.ExtraArgs, fmt.Sprintf("--attrs=node%d", i))
+		c.Start(ctx, t.L(), startOpts, install.MakeClusterSettings(), c.Node(i))
+	}
+
+	database := "tpcc"
+	importCmd := fmt.Sprintf(`./cockroach workload fixtures import tpcc --warehouses=%d`, tc.warehouses)
+	workloadCmd := fmt.Sprintf("./workload run tpcc --warehouses=%d --duration=%s --histograms=%s/stats.json --tolerate-errors {pgurl:1-%d}", tc.warehouses, testTimeout, t.PerfArtifactsDir(), tc.nodes)
+	t.Status(fmt.Sprintf("initializing cluster with %d warehouses", tc.warehouses))
+	c.Run(ctx, c.Node(pinnedNode), importCmd)
```

Comment:
Perhaps also worth plotting how long this import takes, I imagine it's a non-trivial portion of an individual run.